Base classes

Ian-M-M opened the pull request.

Ian-M-M committed:
Restructure files, adding subclasses

Ian-M-M committed:
refactor(barbarian): The barbarian table and bettter explanation for Primal Knowledge

Now the table is formatted.

Primal knowledge was divided into Primal knoledge 1 and 2, to have a better visual reading of the table and progression.

Ian-M-M committed:
feat(barbarian): Add Reckless Skill

This trait is aimed to help the barbarian to have more options outside of combats by espending rages.

Ian-M-M committed:
refactor(bard): Refactor table and Feat name.

Ian-M-M committed:
feat(bard): Spellcasting now have acces ot others spell list.

Now the bard have access to the rest of the spell lists as pseudo half caster, only for preparing spells.

Ian-M-M committed:
refactor(cleric): Table and Feat name.

Ian-M-M committed:
refactor(druid): Table and Feat.

Ian-M-M committed:
feat(fighter): Fighting Feat

Now on addition to the Feat, the fighter can choose to pick a fighting style or maneuver.

Refactor the table and Feat.

Ian-M-M committed:
refactor(monk): Table and Feat.

PJMSGIT reviewed (approved):
[Me parece correcto](https://youtu.be/jhcu6hW2P3s?t=3)

Ian-M-M committed:
refactor(paladin): Table and Feat.

Ian-M-M committed:
feat(paladin): Divine Smite is no longer determined by spells slots.

Now the uses is limited by the proficiency bonus and the damage depends on the higher spell level available.

Ian-M-M committed:
refactor(rogue): Table and Feat.

Ian-M-M committed:
refactor(sorcerer): Table and Feat.

Ian-M-M committed:
refactor(warlock): Table and Feat.

Ian-M-M committed:
refactor(wizard): Table and Feat.

Ian-M-M committed:
feat(wizard): School Speciality.

Divided the school savant into school speciality to have a better progression.

Ian-M-M committed:
refactor(artificer): Table and Feat.

Ian-M-M committed:
refactor(ranger): Table and Feat.

PJMSGIT reviewed (approved):
En general, me gusta como ha quedado para una versión 0.5

hectorhcdev reviewed (approved):
Lo veo todo correcto
